helper/include/odp/helper/chksum.h

```diff
@@ -40,16 +40,16 @@ typedef enum {
  * @param buffer calculate chksum for buffer
  * @param len    buffer length
  *
- * @return checksum value in host cpu order
+ * @return checksum value in network order
  */
 static inline odp_u16sum_t odph_chksum(void *buffer, int len)
 {
-	uint16_t *buf = (uint16_t *)buffer;
+	odp_u16be_t *buf = (odp_u16be_t *)buffer;
 	uint32_t sum = 0;
 	uint16_t result;
 
 	for (sum = 0; len > 1; len -= 2)
-		sum += *buf++;
+		sum += odp_be_to_cpu_16(*buf++);
```

Comment:
I'm not sure why this is necessary. Isn't it more efficient to calculate the checksum directly and then return the result in network byte order? The value is going to be the same, it's just how it's being returned.

Reply:
Hmm, true. I updated this commit to just change documentation.